### PR TITLE
Add withdrawal message to x/rollup

### PR DESCRIPTION
**Pull request by natebeauregard**
- Adds an `InitiateWithdrawal` RPC endpoint to `x/rollup` that takes in a `InitiateWithdrawalRequest` protobuf. The fields from this withdrawal message will also be used by monomer to call the `L2ToL1MessagePasser.initiateWithdrawal` contract after the transaction is executed on the Cosmos SDK appchain and the block is finalized.
- Handler logic for `InitiateWithdrawalRequest` will be done in a follow-up PR 
- Unfortunately we can't use the [Cosmos SDK naming convention](https://docs.cosmos.network/main/build/building-modules/messages-and-queries) because the monomer linter settings are different than traditional cosmos SDK apps

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `MsgInitiateWithdrawal` message type for handling withdrawal transactions.
	- Added an `InitiateWithdrawal` RPC method to facilitate withdrawal processes.
  
- **Enhancements**
	- Renamed and restructured message types for improved clarity, including renaming `ApplyL1TxsRequest` to `MsgApplyL1Txs`.
	- Updated service method signatures to reflect new message types, enhancing consistency across the API.

- **Bug Fixes**
	- Ensured that naming changes do not affect existing functionality, maintaining original method logic.

- **Chores**
	- Added dependency on `cosmos-proto` for enhanced protocol definitions and functionality.

### CODE REVIEW

File: x/rollup/keeper/msg_server.go [hunk 1]
```diff
@@ -101,7 +101,12 @@ func (k *Keeper) ApplyL1Txs(goCtx context.Context, msg *rollupv1.ApplyL1TxsReque
 			return nil, types.WrapError(types.ErrMintETH, "failed to mint ETH", "polymerAddress", cosmAddr, "err", err)
 		}
 	}
-	return &rollupv1.ApplyL1TxsResponse{}, nil
+	return &rollupv1.MsgApplyL1TxsResponse{}, nil
+}
+
+// TODO: implement InitiateWithdrawal handler
+func (k *Keeper) InitiateWithdrawal(goCtx context.Context, msg *rollupv1.MsgInitiateWithdrawal) (*rollupv1.MsgInitiateWithdrawalResponse, error) {
+	return &rollupv1.MsgInitiateWithdrawalResponse{}, nil
```
[the remark this answers is not in the document]
Comment (natebeauregard) [reply x/rollup/keeper/msg_server.go]: nah i'm good

File: proto/rollup/v1/msg.proto [hunk 2]
```diff
@@ -14,5 +20,26 @@ message ApplyL1TxsRequest {
...
 }
 
-message ApplyL1TxsResponse {
+message ApplyL1TxsResponse {}
+
+// MsgInitiateWithdrawal defines a message for all L2 withdrawal txs
+message InitiateWithdrawalRequest {
+  option (cosmos.msg.v1.signer) = "sender";
+  // The cosmos address of the user who wants to withdraw from L2.
+  string sender = 1;
```
Comment (joshklop) [inline proto/rollup/v1/msg.proto:29]: is there a more descriptive type for addresses?

**Review by joshklop — APPROVED** [review]

File: proto/rollup/v1/msg.proto [hunk 3]
```diff
@@ -14,5 +20,26 @@ message ApplyL1TxsRequest {
...
+  // The ethereum address on L1 that the user wants to withdraw to.
+  string target = 2;
+  // The amount of ETH that the user wants to withdraw.
+  string amount = 3 [
```
[the remark this answers is not in the document]
Comment (natebeauregard) [reply proto/rollup/v1/msg.proto:33]: It's converted to a `Math.Int` behind the scenes in [`msg.pb.go`](https://github.com/polymerdao/monomer/pull/69/files#diff-572710651aafcc896b6b77a0e0decdb5ae26656446a1b19f1b9bf37adf0509aeR125) with the `gogoproto.customtype` option

File: proto/rollup/v1/msg.proto (hunk 2 above)
Comment (natebeauregard) [reply proto/rollup/v1/msg.proto:29]: Looks like I can add `[(cosmos_proto.scalar) = "cosmos.AddressString"];` for an auto conversion, I'll add this in

**Pull request merged**